src/features/posts.ts

```diff
@@ -0,0 +1,53 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type PostState = {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const loadPosts = createAsyncThunk(
+  'posts/fetch',
+  async (userId: number) => {
+    const result = await getUserPosts(userId);
+
+    return result;
+  },
+);
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    clearPosts: state => {
+      return { ...state, posts: [] };
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(loadPosts.pending, state => {
+      return { ...state, loaded: false };
+    });
+
+    builder.addCase(loadPosts.fulfilled, (state, actions) => {
+      return { ...state, items: actions.payload };
+    });
+
+    builder.addCase(loadPosts.rejected, state => {
+      return { ...state, hasError: true };
+    });
```

Comment:
In the `loadPosts.rejected` reducer, you are setting `hasError` to `true` when an error occurs. However, you are not resetting `hasError` to `false` when a request is successful. This means that once an error occurs, `hasError` will always be `true`, even if subsequent requests are successful. To fix this, you should set `hasError` to `false` in the `loadPosts.fulfilled` reducer.